extension/awsecshealthcheckextension/extension_test.go

```diff
@@ -0,0 +1,169 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsecshealthcheckextension
+
+import (
+	"context"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"go.opencensus.io/stats/view"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/confignet"
+)
+
+func TestHealthCheckExtensionUsage(t *testing.T) {
+	testEndpoint := "localhost:13134"
+	config := Config{
+		TCPAddr: confignet.TCPAddr{
+			Endpoint: testEndpoint,
+		},
+		Interval:           "5m",
+		ExporterErrorLimit: 1,
+	}
+
+	hcExt := newServer(config, zap.NewNop())
+	require.NotNil(t, hcExt)
+
+	require.NoError(t, hcExt.Start(context.Background(), componenttest.NewNopHost()))
+	t.Cleanup(func() { require.NoError(t, hcExt.Shutdown(context.Background())) })
+
+	// Give a chance for the server goroutine to run.
+	runtime.Gosched()
+
+	newView := view.View{Name: exporterFailureView}
+
+	currentTime := time.Now()
+	vd1 := &view.Data{
+		View:  &newView,
+		Start: currentTime.Add(-2 * time.Minute),
+		End:   currentTime,
+		Rows:  nil,
+	}
+	vd2 := &view.Data{
+		View:  &newView,
+		Start: currentTime.Add(-1 * time.Minute),
+		End:   currentTime,
+		Rows:  nil,
+	}
+
+	hcExt.exporter.exporterErrorQueue = append(hcExt.exporter.exporterErrorQueue, vd1)
+	req1, err := http.NewRequest("GET", testEndpoint, nil)
+	require.Nil(t, err)
+	responseRecorder1 := httptest.NewRecorder()
+	hcExt.handler().ServeHTTP(responseRecorder1, req1)
+	require.Equal(t, http.StatusOK, responseRecorder1.Code)
+
+	hcExt.exporter.exporterErrorQueue = append(hcExt.exporter.exporterErrorQueue, vd2)
+	req2, err := http.NewRequest("GET", testEndpoint, nil)
+	require.Nil(t, err)
+	responseRecorder2 := httptest.NewRecorder()
+	hcExt.handler().ServeHTTP(responseRecorder2, req2)
+	require.Equal(t, http.StatusInternalServerError, responseRecorder2.Code)
+}
+
+func TestHealthCheckExtensionPortAlreadyInUse(t *testing.T) {
+	endpoint := "localhost:13134"
```

Comment:
This has potential to cause bind issues since it matches above, would be better to use `httptest.NewServer` ?